### PR TITLE
[Woo POS] Appearance animation of the products in the cart

### LINKED ISSUE #12344: [Woo POS] Add fade in animation on a new product that appears in the cart

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #12344

### Description

The PR adds fade-in animation to the appearance of a new product in the cart. Compose seems can not animate properly shadows, so I added some softening to the elevation to

### Images/gif

https://github.com/user-attachments/assets/a3fa606f-c12e-4052-ad14-947f3b86828a

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartScreen.kt [hunk 1]
```diff
@@ -316,61 +321,81 @@ private fun ProductItem(
...
+
+                    IconButton(
+                        onClick = { onRemoveClicked(item) },
+                        modifier = Modifier
+                            .size(24.dp)
+                    ) {
+                        Icon(
+                            painter = painterResource(id = R.drawable.ic_pos_remove_cart_item),
+                            tint = MaterialTheme.colors.onBackground,
+                            contentDescription = "Remove item",
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartScreen.kt]: 💡 Let's extract to res

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartScreen.kt [hunk 2]
```diff
@@ -316,61 +321,81 @@ private fun ProductItem(
...
-                    overflow = TextOverflow.Ellipsis
+                AsyncImage(
+                    model = ImageRequest.Builder(LocalContext.current)
+                        .data(item.imageUrl)
+                        .crossfade(true)
+                        .build(),
+                    fallback = ColorPainter(WooPosTheme.colors.loadingSkeleton),
+                    error = ColorPainter(WooPosTheme.colors.loadingSkeleton),
+                    placeholder = ColorPainter(WooPosTheme.colors.loadingSkeleton),
+                    contentDescription = stringResource(R.string.woopos_product_image_description),
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartScreen.kt]:
```suggestion
```
💡 I think we [decided](https://github.com/woocommerce/woocommerce-android/pull/12332/files#diff-fd30e2357047cd3873c2b6cf479ab48081a5165511c6abce9f2b055a14db2193L337) to remove it.

**Review by samiuelson — COMMENTED** [review]
Thanks for your work, @kidinov! I've added two small code suggestions. Please, take a look.

Additionally, I tested the UX a bit, to check the performance. Some of the issues below might have existed before animations. However, animation and its delay/duration don't help or make things worse. Take a look:


|Issues|
|---|
|1. It's hard to scroll if the cart is long; sometimes animations are disturbing during the scroll|
|<video src="https://github.com/user-attachments/assets/b6a00021-bffe-482f-871c-9d8128f24712"/>|
|2. The Cart doesn't keep up rendering with continuous fast taps on items in product selector|
|<video src="https://github.com/user-attachments/assets/01234e78-d01b-45d6-8b43-43779a67398e"/>|
|3. nit-pick: After config change animation is replayed|
|<video src="https://github.com/user-attachments/assets/f9bb70c3-d1a3-4653-930b-80b878d3f39c"/>|

Let me know what you think!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartScreen.kt (hunk 1 above)
Comment (samiuelson) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartScreen.kt]: Actually, let's remove the content description from here. It's been deleted already and replaced by more detailed one in IconButton [here](https://github.com/woocommerce/woocommerce-android/pull/12332/files#diff-fd30e2357047cd3873c2b6cf479ab48081a5165511c6abce9f2b055a14db2193R388)
```suggestion
```

Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartScreen.kt]: There are no changes regarding this in this PR, except intends. I updated it from the trunk to make it less confusing

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartScreen.kt (hunk 2 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartScreen.kt]: The same here https://github.com/woocommerce/woocommerce-android/pull/12352#discussion_r1721556774

**Comment by kidinov** [comment]
@samiuelson 👋

You may have tested on the debug build, and performance is pretty bad then in Compose UI, so it is better to test things like that in the release

https://developer.android.com/develop/ui/compose/performance#config

Having said that, it seems that I have to fix those anyway, and the animation flag should be part of the state in the VM. I'll move it, thanks 🙏

**Comment by kidinov** [comment]
@samiuelson I updated the animation and kept its state in the VM. Please take a look

It looks "ok," but not too good, in my opinion

I am wondering what your opinion on it:
* Shell, we keep the animation at all or maybe forget about this as this brings quite some extra complexity
* Maybe try another animation?
* Or do you feel it's ok how it's now?

https://github.com/user-attachments/assets/dca01606-6a9c-4d05-9b84-3531c0bd9153

**Comment by samiuelson** [comment]
> @samiuelson I updated the animation and kept its state in the VM. Please take a look
> 
> It looks "ok," but not too good, in my opinion
> 
> I am wondering what your opinion on it:
> * Shell, we keep the animation at all or maybe forget about this as this brings quite some extra complexity
> * Maybe try another animation?
> * Or do you feel it's ok how it's now?

👋🏼 @kidinov, I've tested a release build and can confirm the performance is much better – no delay 👍🏼 The state is also preserved during config change, so I think it's in an acceptable shape for merge. 

Maybe reducing delays and durations would help a bit 🤔 We can also ask for design feedback if it's better to have animations in the cart or not – up to you.

**Review by samiuelson — APPROVED** [review]
LGTM!

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/232">20.1</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**